java/pubsub/src/main/java/com/example/pubsub/OpenCensusStatsUtil.java

```diff
@@ -0,0 +1,135 @@
+/* Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.pubsub;
+
+import com.google.cloud.ServiceOptions;
+
+import io.opencensus.common.Duration;
+import io.opencensus.common.Scope;
+import io.opencensus.exporter.stats.stackdriver.StackdriverStatsConfiguration;
+import io.opencensus.exporter.stats.stackdriver.StackdriverStatsExporter;
+import io.opencensus.stats.Aggregation.Distribution;
+import io.opencensus.stats.BucketBoundaries;
+import io.opencensus.stats.Measure;
+import io.opencensus.stats.Stats;
+import io.opencensus.stats.StatsRecorder;
+import io.opencensus.stats.View;
+import io.opencensus.tags.TagContext;
+import io.opencensus.tags.TagKey;
+import io.opencensus.tags.TagValue;
+import io.opencensus.tags.Tagger;
+import io.opencensus.tags.Tags;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * A utility class for using OpenCensus stats with pub/sub.
+ */
+final class OpenCensusStatsUtil {
+  private static final Tagger tagger = Tags.getTagger();
+
+  private static final StatsRecorder statsRecorder = Stats.getStatsRecorder();
+
+  private static final String PROJECT_ID = ServiceOptions.getDefaultProjectId();
```

Comment:
`ServiceOptions.getDefaultProjectId()` could return null, consider checking null project ID here.

Reply:
Done - moved code to the static initializer.